File: front/src/app/services/message.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Message} from '../modules/message/model/message';

@Injectable({
  providedIn: 'root'
})
export class MessageService {

  constructor(private  http: HttpClient) {
  }

  getAll(): Observable<Message[]> {
    return this.http.get<Message[]>('/api/messages/all');
  }

  deleteMessage(idMessage: string) {
    return this.http.delete('/api/messages/' + idMessage);
  }

  saveMessage(message: Message) {
    return this.http.post('/api/messages', message);
  }

  getById(idMessage: string): Observable<Message> {
    return this.http.get<Message>('/api/messages/id/' + idMessage);
  }

  getByReceiver(login: string): Observable<Message[]> {
    return this.http.get<Message[]>('/api/messages/byReceiver/' + login);
  }

  saveAnyMessages(messages: Message[]) {
    return this.http.post('/api/messages/any', messages);
  }
}

File: fapi/src/main/java/fapi/service/MessageService.java
package fapi.service;

import fapi.models.Message;
import fapi.models.User;

import java.util.List;
import java.util.Optional;

public interface MessageService {
    Message saveMessage(Message message);

    Message findMessageById(Long id);

    List<Message> findAll();

    void deleteMessage(Long id);

    List<Message> findAllByReceiver(String login);

    void saveAnyMessages (Message[] messages);
}
